plugins/osdn/factory/factory.go

```diff
@@ -3,31 +3,31 @@ package factory
 import (
 	"strings"
 
-	"github.com/openshift/openshift-sdn/plugins/osdn"
-	"github.com/openshift/openshift-sdn/plugins/osdn/api"
-
 	osclient "github.com/openshift/origin/pkg/client"
 	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/storage"
 
+	"github.com/openshift/openshift-sdn/plugins/osdn"
+	"github.com/openshift/openshift-sdn/plugins/osdn/api"
 	"github.com/openshift/openshift-sdn/plugins/osdn/ovs"
 )
 
 // Call by higher layers to create the plugin SDN master instance
-func NewMasterPlugin(pluginType string, osClient *osclient.Client, kClient *kclient.Client) (api.OsdnPlugin, error) {
-	return newPlugin(pluginType, osClient, kClient, "", "")
+func NewMasterPlugin(pluginType string, osClient *osclient.Client, kClient *kclient.Client, etcdHelper storage.Interface) (api.OsdnPlugin, error) {
```

Comment:
entangling the plugin with etcd and migration seems less than ideal. Shouldn't the migration responsibility live where the registry was previously, and the plugin should continue interacting only with the API?